tests/v2/test_waku_noise.nim

```diff
@@ -0,0 +1,6 @@
+{.used.}
+
+import
+  testutils/unittests,
```

Comment:
the test file looks empty, is it intentional?

Reply:
Yes it was intentionally left blank since this PR comes from a process of cherry picking commits to split the big WIP implementation into smaller PRs. All changes to test_waku_noise.nim can in fact be discarded from this PR, but if we quickly merge this and all the other noise related PR then would not be necessary.

Reply:
Not important to address in this PR, but in future I think it's preferable having each incremental PR be reasonable as a stand-alone that can be reviewed in its entirety. It makes sense to have open TODOs, of course, but since empty modules can't really be reviewed we increase the risk of having dead code in future that we forget about.